plugins/flexibleengine/src/main/java/org/eclipse/xpanse/plugins/flexibleengine/manage/FlexibleEngineResourceManager.java

```diff
@@ -98,130 +99,159 @@ public List<String> getExistingResourceNamesWithKind(String userId,
      * @return availability zones
      */
     public List<String> getAvailabilityZonesOfRegion(String userId, String region) {
-        List<String> availabilityZones = new ArrayList<>();
+        List<String> availabilityZoneNames = new ArrayList<>();
         try {
             EcsClient ecsClient = getEcsClient(userId, region);
+            NovaListAvailabilityZonesRequest request = new NovaListAvailabilityZonesRequest();
             NovaListAvailabilityZonesResponse response =
-                    ecsClient.novaListAvailabilityZones(new NovaListAvailabilityZonesRequest());
+                    ecsClient.novaListAvailabilityZonesInvoker(request)
+                            .retryCondition(flexibleEngineClient::matchRetryCondition)
+                            .backoffStrategy(new FlexibleEngineRetryStrategy(DEFAULT_DELAY_MILLIS))
+                            .invoke();
             if (response.getHttpStatusCode() == 200) {
-                availabilityZones = response.getAvailabilityZoneInfo().stream()
+                availabilityZoneNames = response.getAvailabilityZoneInfo().stream()
                         .map(NovaAvailabilityZone::getZoneName).toList();
             }
-        } catch (SdkException e) {
-            log.error("Get FlexibleEngine availability zones of region:{} failed, error:{}",
-                    region, e.getMessage());
+        } catch (Exception e) {
+            log.error("FlexibleEngineClient listAvailabilityZones with region:{} error.", region,
```

Comment:
@baixinsui i'm a little confused. here the problem is still the same know? we still just catch all exceptions and return an empty list. My expectation is that we throw an exception to client if this call fails.

Reply:
Ok.I misunderstood. I will throw the ClientApiCallFailedException(errorMsg)